### PR TITLE
add nlp tools

### PR DESCRIPTION
**Pull request by zhimin-z**
add Guardrails

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -239,7 +239,6 @@ This repository contains a curated list of awesome open source libraries that wi
...
 * [keras-tuner](https://github.com/keras-team/keras-tuner) ![](https://img.shields.io/github/stars/keras-team/keras-tuner?style=social) - Keras Tuner is an easy-to-use, distributable hyperparameter optimization framework that solves the pain points of performing a hyperparameter search. Keras Tuner makes it easy to define a search space and leverage included algorithms to find the best hyperparameter values.
 * [Maggy](https://github.com/logicalclocks/maggy) ![](https://img.shields.io/github/stars/logicalclocks/maggy.svg?style=social) - Asynchronous, directed Hyperparameter search and parallel ablation studies on Apache Spark - [(Video)](https://www.youtube.com/watch?v=0Hd1iYEL03w).
-* [mljar-supervised](https://github.com/mljar/mljar-supervised) ![](https://img.shields.io/github/stars/mljar/mljar-supervised.svg?style=social) - An Automated Machine Learning (AutoML) python package for tabular data. It can handle: Binary Classification, MultiClass Classification and Regression. It provides feature engineering, explanations and markdown reports.
```
Comment (axsaucedo) [inline README.md:242]: Seems removed by mistake

File: README.md [hunk 2]
```diff
@@ -468,10 +467,12 @@ This repository contains a curated list of awesome open source libraries that wi
...
 * [Flair](https://github.com/zalandoresearch/flair) ![](https://img.shields.io/github/stars/zalandoresearch/flair.svg?style=social) - Simple framework for state-of-the-art NLP developed by Zalando which builds directly on PyTorch.
 * [FlexGen](https://github.com/FMInference/FlexGen) ![](https://img.shields.io/github/stars/FMInference/FlexGen.svg?style=social) - FlexGen is a high-throughput generation engine for running large language models with limited GPU memory.
-* [Github's Semantic](https://github.com/github/semantic) ![](https://img.shields.io/github/stars/github/semantic.svg?style=social) - Github's text library for parsing, analyzing, and comparing source code across many languages .
+* [Github's Semantic](https://github.com/github/semantic) ![](https://img.shields.io/github/stars/github/semantic.svg?style=social) - Guardrails is a Python package for specifying structure and type, validating and correcting the outputs of large language models (LLMs).
```
Comment (axsaucedo) [inline README.md:470]: Seems error

File: README.md [hunk 3]
```diff
@@ -468,10 +467,12 @@ This repository contains a curated list of awesome open source libraries that wi
...
 * [GPT Index (LlamaIndex)](https://github.com/jerryjliu/gpt_index) ![](https://img.shields.io/github/stars/jerryjliu/gpt_index.svg?style=social) - GPT Index (LlamaIndex) is a project consisting of a set of data structures designed to make it easier to use large external knowledge bases with LLMs.
+* [Gretel Synthetics](https://github.com/gretelai/gretel-synthetics) ![](https://img.shields.io/github/stars/gretelai/gretel-synthetics.svg?style=social) - Gretel Synthetics is a synthetic data generators for structured and unstructured text, featuring differentially private learning.
 * [Grover](https://github.com/rowanz/grover) ![](https://img.shields.io/github/stars/rowanz/grover.svg?style=social) - Grover is a model for Neural Fake News -- both generation and detection. However, it probably can also be used for other generation tasks.
+* [Guardrails](https://github.com/ShreyaR/guardrails) ![](https://img.shields.io/github/stars/ShreyaR/guardrails.svg?style=social) - Github's text library for parsing, analyzing, and comparing source code across many languages.
```
Comment (axsaucedo) [inline README.md:475]: Error here as well

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Comments on a few errors

File: README.md (hunk 2 above)
Comment (zhimin-z) [reply README.md]: Done! @axsaucedo

**Pull request merged**